WordPress/src/main/java/org/wordpress/android/ui/people/PeopleListFragment.java

```diff
@@ -72,16 +93,120 @@ public void refreshPeopleList() {
-        if (mListener != null) {
+        if (mOnPersonSelectedListener != null) {
             Person person = (Person) parent.getItemAtPosition(position);
-            mListener.onPersonSelected(person);
+            mOnPersonSelectedListener.onPersonSelected(person);
         }
     }
 
     // Container Activity must implement this interface
     public interface OnPersonSelectedListener {
         void onPersonSelected(Person person);
     }
+
+    public interface OnFetchMorePeopleListener {
+        void onFetchMorePeople();
+    }
+
+    public class PeopleAdapter extends BaseAdapter {
+        private final LayoutInflater mInflater;
+        private List<Person> mPeopleList;
+        private int mAvatarSz;
+
+        public PeopleAdapter(Context context, List<Person> peopleList) {
+            mAvatarSz = context.getResources().getDimensionPixelSize(R.dimen.people_avatar_sz);
+            mInflater = LayoutInflater.from(context);
+            mPeopleList = peopleList;
+        }
+
+        public void setPeopleList(List<Person> peopleList) {
+            mPeopleList = peopleList;
+            notifyDataSetChanged();
+        }
+
+        @Override
+        public int getCount() {
+            if (mPeopleList == null) {
+                return 0;
+            }
+            return mPeopleList.size();
+        }
+
+        @Override
+        public Person getItem(int position) {
+            if (mPeopleList == null) {
+                return null;
+            }
+            return mPeopleList.get(position);
+        }
+
+        @Override
+        public long getItemId(int position) {
+            Person person = getItem(position);
+            if (person == null) {
+                return -1;
+            }
+            return person.getPersonID();
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            final PeopleViewHolder holder;
+
+            if (convertView == null || convertView.getTag() == null) {
+                convertView = mInflater.inflate(R.layout.people_list_row, parent, false);
+                holder = new PeopleViewHolder(convertView);
+                convertView.setTag(holder);
+            } else {
+                holder = (PeopleViewHolder) convertView.getTag();
+            }
+
+            Person person = getItem(position);
+
+            if (person != null) {
+                String avatarUrl = GravatarUtils.fixGravatarUrl(person.getAvatarUrl(), mAvatarSz);
+                holder.imgAvatar.setImageUrl(avatarUrl, WPNetworkImageView.ImageType.AVATAR);
+                holder.txtDisplayName.setText(person.getDisplayName());
+                holder.txtUsername.setText(String.format("@%s", person.getUsername()));
+                holder.txtRole.setText(StringUtils.capitalize(person.getRole()));
+            }
+
+            // end of list is reached
+            if (mOnFetchMorePeopleListener != null && position == getCount() - 1) {
```

Comment:
This (or the related code in the activity) should prevent fetching users when a fetch is already in progress. Otherwise the user could scroll up and down, causing multiple fetches for the same offset.